### PR TITLE
android fix

### PR DESCRIPTION
**Pull request by sadiuk**
I left the void* instead of the typedefs cause i couldn't find the definitions for them (they are defined in egl.h, but compiler still won't recognize them).

### CODE REVIEW

File: EGL/include/EGL/eglctxinternals.h [hunk 1]
```diff
@@ -29,7 +28,13 @@ struct _EGLContextInternals
     } surface;
     GLXContext context;
 };
-
+#elif defined(__ANDROID__)
+#include "egl.h"
```
Comment (Crisspl) [inline EGL/include/EGL/eglctxinternals.h]: i dont think you can include egl.h here, i think it circular dependency

File: EGL/include/EGL/eglctxinternals.h [hunk 2]
```diff
@@ -14,8 +14,7 @@ struct _EGLContextInternals
     HGLRC context;
 };
 
-#elif defined(__unix__)
-
+#elif defined(X_PROTOCOL)
```
Comment (Crisspl) [inline EGL/include/EGL/eglctxinternals.h]: X_PROTOCOL? where this define come from?

File: EGL/include/EGL/eglctxinternals.h [hunk 3]
```diff
@@ -29,7 +28,13 @@ struct _EGLContextInternals
     } surface;
     GLXContext context;
 };
-
+#elif defined(__ANDROID__)
+#include "egl.h"
+struct _EGLContextInternals
+{
+    void* surface; // EGLSurface
```
Comment (Crisspl) [inline EGL/include/EGL/eglctxinternals.h:35]: just

File: EGL/include/EGL/eglctxinternals.h (hunk 1 above)
Comment (Crisspl) [reply EGL/include/EGL/eglctxinternals.h]: but if it is possible then you don't have to void*, you can just use EGLcontext and EGLSurface then

Comment (sadiuk) [reply EGL/include/EGL/eglctxinternals.h]: yeah it's a circular dep, forgot to remove that include

**Comment by Crisspl** [comment]
ok, now define `eglGetPlatformDependentHandles` for passthrough backend (in passthrough.c)
it should be just something like this
```
EGLAPI EGLBoolean EGLAPIENTRY eglGetPlatformDependentHandles (EGLContextInternals* out, EGLDisplay dpy, EGLSurface surface, EGLContext ctx)
{
	out.context = ctx;
        out.surface = surface;
return EGL_TRUE;
}
```
and you're done

**Pull request merged**